src/workload/mod.rs

```diff
@@ -377,7 +377,8 @@ impl Generator {
             Verb::SetIfPresentAndNotEqual => {
                 ClientRequest::SetIfPresentAndNotEqual(client::SetIfPresentAndNotEqual {
                     key: keyspace.sample(rng),
-                    value: keyspace.gen_value(sequence as _, rng),
+                    new_value: keyspace.gen_value(sequence as _, rng),
+                    old_value: keyspace.gen_value(sequence as _, rng),
```

Comment:
If you generate different values for old_value and new_value, the result will always be `stored`. Is that intentional?

Reply:
There might not be a good alternative. We don't really have any place we're keeping history for each key to do some % of overwrite. Similar problem as testing something like a CAS operation. You might be able to fake something with two keyspaces? But the workload generation isn't really designed with this in mind. 

If there's need in the future, what I'd consider is some mechanism for playing workload from a file and trying to make that flexible in terms of whether exact or random out of N values are used in each place. Then you could dial in some very specific patterns by generating that file or getting some prod trace and replaying it.

For now, maybe a comment acknowledging this behavior and that it's expected would be good?

Reply:
sounds good